packages/flutter_tools/test/commands.shard/hermetic/test_test.dart

```diff
@@ -59,6 +59,52 @@ void main() {
     Cache: () => FakeCache(),
   });
 
+  group('shard-index and total-shards', () {
+    testUsingContext('with the params they are Piped to package:test',
+        () async {
+      final FakePackageTest fakePackageTest = FakePackageTest();
+
+      final TestCommand testCommand = TestCommand(testWrapper: fakePackageTest);
+      final CommandRunner<void> commandRunner =
+          createTestCommandRunner(testCommand);
+
+      await commandRunner.run(const <String>[
+        'test',
+        '--total-shards=1',
+        '--shard-index=2',
+        '--no-pub',
+      ]);
+
+      expect(fakePackageTest.lastArgs, contains('--total-shards=1'));
+      expect(fakePackageTest.lastArgs, contains('--shard-index=2'));
+    }, overrides: <Type, Generator>{
+      FileSystem: () => fs,
+      ProcessManager: () => FakeProcessManager.any(),
+      Cache: () => FakeCache(),
+    });
+
+    testUsingContext('without the params they not Piped to package:test',
+        () async {
+      final FakePackageTest fakePackageTest = FakePackageTest();
+
+      final TestCommand testCommand = TestCommand(testWrapper: fakePackageTest);
+      final CommandRunner<void> commandRunner =
+          createTestCommandRunner(testCommand);
+
+      await commandRunner.run(const <String>[
+        'test',
+        '--no-pub',
+      ]);
+
+      expect(fakePackageTest.lastArgs, isNot(contains('--total-shards=1')));
```

Comment:
one last nit, could you make these patterns not include the hard-coded numbers, i.e. `isNot(contains('--total-shards'))`. After that LGTM, and we can ship this :)

Reply:
done 👍🏻